hawkbit-repository/hawkbit-repository-api/src/main/java/org/eclipse/hawkbit/repository/FilterParams.java

```diff
@@ -107,17 +126,50 @@ public String getFilterBySearchText() {
      *
      * @return the flag indicating if tagging filter is used
      */
-    public Boolean getSelectTargetWithNoTag() {
+    public boolean isSelectTargetWithNoTag() {
         return selectTargetWithNoTag;
     }
 
     /**
      * Gets the tags that are used to filter for. The activation of this filter
-     * is done by {@link #setSelectTargetWithNoTag(Boolean)}.
+     * is done by {@link #isSelectTargetWithNoTag()}.
      *
-     * @return the tags that are used to filter for
+     * @return the tags that are used to filter for or an empty array if non are
+     *         present
      */
     public String[] getFilterByTagNames() {
         return filterByTagNames;
     }
+
+    /**
+     * checks if there is a valid filter specified
+     * 
+     * @return true if there is no valid filter set
+     */
+    public boolean isEmpty() {
+        return isEmpty(filterByDistributionId);
+    }
+
+    /**
+     * Checks if there is a valid filter set. The filter is empty if no filter
+     * criteria is set.
+     * 
+     * @param filterByDistributionId
+     *            {@link DistributionSet#getId()}
+     * @return true if there is no valid filter set.
+     */
+    public boolean isEmpty(Long filterByDistributionId) {
+        return filterByStatus.isEmpty() || getFilterBySearchText().isPresent() || filterByDistributionId != null
```

Comment:
double check semantics, if `getFilterBySearchText().isPresent()`, is the `FilterParams` considered empty or not?
as well as `filterByDistributionId != null`